census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,338 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""This code implements a Feed forward neural network using TF low level APIs.
+   It implements a binary classifier for Census Income Dataset using both single
+   and distributed node cluster. 
+"""
+
+
+import argparse
+import json
+import tensorflow as tf
+import numpy as np
+import pandas as pd
+from tensorflow.contrib.layers.python.ops import sparse_feature_cross_op
+from tensorflow.contrib.layers.python.ops import bucketization_op
+
+from StringIO import StringIO
+
+import model
+import os
+
+tf.logging.set_verbosity(tf.logging.INFO)
+
+#See tutorial on wide and deep https://www.tensorflow.org/tutorials/wide_and_deep/
+#https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/feature_column.py
+
+# csv columns in the input file
+CSV_COLUMNS = ('age', 'workclass', 'fnlwgt', 'education', 'education_num',
+               'marital_status', 'occupation', 'relationship', 'race', 'gender',
+               'capital_gain', 'capital_loss', 'hours_per_week', 'native_country',
+               'income_bracket')
+
+CATEGORICAL_COLS = ('gender', 'race', 'education', 'marital_status',
+                    'relationship', 'workclass','occupation', 'native_country')
+
+CONTINUOUS_COLS = ('age', 'education_num', 'capital_gain', 'capital_loss', 'hours_per_week')
+
+LABEL_COL = 'income_bracket'
+
+
+#
+# Graph creation section for training and evaluation
+#
+def make_graph(inputs, labels, learning_rate=0.5):
+  """Create training and evaluation graph."""
+
+  logits = model.inference(inputs)
+
+  global_step = tf.contrib.framework.get_or_create_global_step()
+  cross_entropy = tf.reduce_mean(
+      tf.nn.softmax_cross_entropy_with_logits(logits=logits, labels=labels))
+
+  train_step = tf.train.GradientDescentOptimizer(learning_rate).minimize(
+      cross_entropy, global_step=global_step)
+
+  correct_prediction = tf.equal(tf.argmax(logits, 1), tf.argmax(labels, 1))
+  eval_step = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+
+  return train_step, eval_step, global_step
+
+def get_placeholders():
+  """Create placeholder for inputs and prediction labels.
+
+  Returns:
+     tuple (tf.placeholder): 337 dimension feature and 2 classes
+  """
+  inputs = tf.placeholder(tf.float32, shape=[None, 337])
+  labels = tf.placeholder(tf.float32, shape=[None, 2])
+  return inputs, labels
+
+def read_input_data(file_name, skiprows=None):
```

Comment:
I suspect this is in the works -- the pandas DataFrame thing will be going away.

Needs to build the input queue (tf.train.match_files_once, tf.TextLineReader, then building either a shuffle batch or just batch), i.e. do not rely on loading all the data into an in-memory DataFrame. This isn't possible in real-world cases where you have more data.